development/image-builder/main.go

```diff
@@ -387,6 +398,23 @@ func loadEnv(vfs fs.FS, envFile string) (map[string]string, error) {
 	return vars, nil
 }
 
+// Add parsed tags to environments which will be passed to dockerfile
+func addTagsToEnv(tags []string, envs map[string]string) map[string]string {
+	m := make(map[string]string)
+
+	for i, t := range tags {
+		// (@KacperMalachowski): ENV VAR key shouldn't be hardcoded, we have to find way to parametrize it in the future
+		key := fmt.Sprintf("DOCKER_TAG_%d", i)
+		m[key] = t
```

Comment:
Im still not super convenient about this approach. Let's say for example you have multiple tags defined in the runtime. The code itself doesn't sort, so guessing the index of a tag is not intuitive. It will require from the dev to trial and error the actual result, which should not be the case.

Maybe we should introduce named tags. For example instead of providing only a value, we could add parsing of specific key/value field:
```
--tag=name={{ .TagTemplate }}
```
And then `--export-tags` would only export the values that contain this named key.

I cannot accept the feature that it is made right now.

/hold

Reply:
I suggest to keep backward compatibility, introduce something like that:

If a key is defined, then we use that custom key, otherwise we can simply take format.

So for example having 
`--tag {{ .ShortSHA }}` 

will be parsed to:

```
key: ShortSHA
value: dce1515156
```

What do you think?

Reply:
Thinking about key...
I found that it should have a prefix to be unique in environment, so maybe rather something like: 
`TAG_ShortSHA`  instead of just `ShortSHA`.

Reply:
It's not always well defined in the Tagger struct. The use case is also to utilize .Env as tag. To get more control maybe we should deprecate this function and fill them as static values in struct. Then you can easily use that approach.

Reply:
Can you explain more about using .Env as tag? I cannot see it in tagger code. If it's not implemented yet, we can just ignore that, so I can prepare struct for tag naming feature.

Reply:
It is implemented, see Tagger struct https://github.com/kyma-project/test-infra/blob/6cc717f8958ede77a01e0a4ad9ef37a8f24a0370/development/pkg/tags/tags.go#L25

Reply:
Oh, got it. I see that, but I don't know that I can call methods from go template when they need parameters.

It shouldn't be a problem, as the potential DOCKER_TAG_x env variable will be set after all tags are evaluated.
The statement in documentation "You cannot use DOCKER_TAG_x variables as docker tag" should be enough in that case.